src/app/courses/courses.module.ts

```diff
@@ -23,7 +28,13 @@ import { MaterialModule } from '../shared/material.module';
   declarations: [
     CoursesComponent,
     CoursesAddComponent,
-    CoursesRequestComponent
-  ]
+    CoursesRequestComponent,
+    CourseOpenComponent,
+    CourseManageComponent,
+    CourseViewComponent,
+    CourseCreditsComponent,
+    CourseProgressComponent
+  ],
+  providers: [ CourseValidatorService ]

```

Comment:
Course validator is now changed to ValidatorService please change it, and do import validator service